layer/nrlf/core/validators.py

```diff
@@ -316,3 +317,64 @@ def _validate_ssp_asid(self, model: DocumentReference):
                 field=f"context.related[{idx}].identifier.value",
             )
             return
+
+    def _validate_category(self, model: DocumentReference):
+        """
+        Validate the category field contains an appropriate coding system, code and display.
+        """
+
+        if len(model.category) > 1:
+            logger.log(
+                LogReference.VALIDATOR001, step="category", reason="category_too_long"
+            )
+            self.result.add_error(
+                issue_code="invalid",
+                error_code="INVALID_RESOURCE",
+                diagnostics=f"Invalid category length: {len(model.category)}",
+                field=f"category",
+            )
+            return
+
+        logger.log(LogReference.VALIDATOR001, step="category")
+
+        logger.debug("Validating category")
+
+        for index, coding in enumerate(model.category[0].coding):
+            if coding.system != "http://snomed.info/sct":
+                self.result.add_error(
+                    issue_code="value",
+                    error_code="INVALID_RESOURCE",
+                    diagnostics=f"Invalid category system: {coding.system}",
+                    field=f"category[0].coding[{index}].system",
+                )
+                continue
+
+            if coding.code not in [
```

Comment:
could create a global mapping for these codes, maybe something like this:

```
categories = [
  { 
    "code": "734163000",
    "display": "Care plan"
  }, 
  {
    "code": "1102421000000108",
    "display": "Observations"
  }
]
```

just a rough suggestion though, not saying you need to change anything here, I'll leave that up to you.

Reply:
I originally wanted to do somat like this where I had the code as a key and the display as a value and then mapped the input code and display to the key value pair as validation but then thought that because theres only the 2 categories its not really necessary and unless we are expecting a lot more we dont really need to try and map as its just simpler to do it the current way but if we had a lot more categories then this would be a way better way of doing it than individually checking each one.  I might try and implement it to see if its quick tho

Reply:
alright improved the way its done i think lemme know what u think

Reply:
Ah nice! I like that :muscle: :clap: